File: 2024-2/0684/unidad04.d - Statement 02/SoftPubPersonaBO/src/pe/edu/pucp/softpub/persona/bo/EspecialidadBO.java
package pe.edu.pucp.softpub.persona.bo;

import pe.edu.pucp.softpub.persona.dao.EspecialidadDAO;
import pe.edu.pucp.softpub.persona.daoImp.EspecialidadDAOImpl;
import pe.edu.pucp.softpub.persona.model.Especialidad;
import pe.edu.pucp.softpub.persona.model.Nivel;

public class EspecialidadBO {
    
    private EspecialidadDAO especialidadDAO;
    
    public EspecialidadBO(){
        this.especialidadDAO = new EspecialidadDAOImpl();
    }
    
    public Integer insertar(Integer idEspecialidad, String nombre_especialidad, String facultad, Nivel nivel){
        Especialidad especialidad = new Especialidad(idEspecialidad, nombre_especialidad, facultad, nivel);
        return this.especialidadDAO.insertar(especialidad);
    }
}

File: 2024-2/0684/unidad06.d_Web_04/SofPub/SoftPubPersonaBO/EspecialidadBO.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using SoftPubPersonaDA.DAO;
using SoftPubPersonaDA.DAOImpl;
using SoftPubPersonaModel;

namespace SoftPubPersonaBO
{    
    public class EspecialidadBO
    {
        private EspecialidadDAO especialidadDAO;

        public EspecialidadBO()
        {
            this.especialidadDAO = new EspecialidadDAOImpl();
        }

        public int insertar(int idEspecialidad, string nombre_especialidad, string facultad, string descripcion_nivel)
        {
            Nivel nivel = (Nivel)Enum.Parse(typeof(Nivel), descripcion_nivel);
            Especialidad especialidad = new Especialidad(idEspecialidad, nombre_especialidad, facultad, nivel);
            return this.especialidadDAO.insertar(especialidad);
        }

        public int modificar(int idEspecialidad, string nombre_especialidad, string facultad, string descripcion_nivel)
        {
            Nivel nivel = (Nivel)Enum.Parse(typeof(Nivel), descripcion_nivel);
            Especialidad especialidad = new Especialidad(idEspecialidad, nombre_especialidad, facultad, nivel);
            return this.especialidadDAO.modificar(especialidad);
        }

        public int eliminar(int idEspecialidad, string nombre_especialidad, string facultad, string descripcion_nivel)
        {
            Nivel nivel = (Nivel)Enum.Parse(typeof(Nivel), descripcion_nivel);
            Especialidad especialidad = new Especialidad(idEspecialidad, nombre_especialidad, facultad, nivel);
            return this.especialidadDAO.eliminar(especialidad);
        }

        public BindingList<Object> listarTodos()
        {
            return this.especialidadDAO.listarTodos();
        }
    }
}
